src/iceberg/table.cc

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "iceberg/table.h"
+
+#include "iceberg/partition_spec.h"
+#include "iceberg/schema.h"
+#include "iceberg/sort_order.h"
+#include "iceberg/table_metadata.h"
+
+namespace iceberg {
+
+const std::string& Table::uuid() const { return metadata_->table_uuid; }
+
+const std::shared_ptr<Schema>& Table::schema() const {
+  if (!schema_) {
+    const static std::shared_ptr<Schema> kEmptySchema =
+        std::make_shared<Schema>(std::vector<SchemaField>{});
+    auto schema = metadata_->Schema();
+    if (schema.has_value()) {
+      schema_ = schema.value();
+    } else {
+      schema_ = kEmptySchema;
+    }
+  }
+  return schema_;
+}
+
+const std::unordered_map<int32_t, std::shared_ptr<Schema>>& Table::schemas() const {
+  std::call_once(init_schemas_once_, [this]() {
```

Comment:
What about removing the once flag and use `std::shared_ptr<std::unordered_map<int32_t, std::shared_ptr<Schema>>>` instead?